### PR TITLE
Make editor update its associated editor tab value & breakpoints

### PR DESCRIPTION
**Pull request by ianyong**
### Description

The `Editor` component now takes in an editor tab index as prop. It uses the editor tab index to update the value and breakpoints of the editor tab it is associated with.

Part of the refactoring for #2176. Follow-up to #2367.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to test

There should be no observable changes in behaviour since the multiple files mode is still disabled.

### CODE REVIEW

File: src/pages/playground/Playground.tsx [hunk 1]
```diff
@@ -723,8 +722,6 @@ const Playground: React.FC<PlaygroundProps> = ({ workspaceLocation = 'playground
 
   const onChangeMethod = React.useCallback(
     (newCode: string, delta: CodeDelta) => {
-      handleEditorValueChange(newCode);
```
Comment (ianyong) [inline src/pages/playground/Playground.tsx:726]: This results in the editor value being set twice, once in the `Editor` component, and once in the playground workspace. You can verify that this is safe to remove by observing that you can still change the value of the editor in the playground, or by observing that prior to this PR, the Redux action `UPDATE_EDITOR_VALUE` is fired twice for each change.

**Review by zhaojj2209 — COMMENTED** [review]
Just some nits

File: src/commons/editor/__tests__/Editor.tsx [hunk 2]
```diff
@@ -6,15 +6,16 @@ import { Position } from '../EditorTypes';
 
 test('Editor renders correctly', () => {
   const props: EditorProps = {
+    editorTabIndex: 1,
```
[the remark this answers is not in the document]
Comment (ianyong) [reply src/commons/editor/__tests__/Editor.tsx:9]: To be honest, this snapshot test is useless, so it doesn't matter what value is given. You do raise a good point that the other properties use default values though. I'll make the change.

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

**Pull request merged**